functors_task_1/student/include/impl.h

```diff
@@ -8,6 +8,16 @@
  */
 struct BeerOrganizer
 {
+private:
+    size_t currentIndex {1};
+
+public:
+    BeerBrand operator()() {
+        if (currentIndex == static_cast<int>(BeerBrand::Max)) {
+            currentIndex = 1;
+        }
+        return BeerBrand(currentIndex++);
```

Comment:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es49-if-you-must-use-a-cast-use-a-named-cast

Reply:
Done